src/sm/launcher/launcher.cpp

```diff
@@ -114,21 +134,21 @@ Error Launcher::Stop()
 Error Launcher::RunInstances(const Array<ServiceInfo>& services, const Array<LayerInfo>& layers,
     const Array<InstanceInfo>& instances, bool forceRestart)
 {
-    UniqueLock lock {mMutex};
-
-    if (forceRestart) {
-        LOG_DBG() << "Restart instances";
-    } else {
-        LOG_DBG() << "Run instances";
-    }
+    {
+        LockGuard lock {mMutex};
 
-    if (mLaunchInProgress) {
-        return AOS_ERROR_WRAP(ErrorEnum::eWrongState);
-    }
+        if (forceRestart) {
+            LOG_DBG() << "Restart instances";
```

Comment:
Is it possible to do in one line?
```suggestion
            LOG_DBG() << (forceRestart) ? "Restart instances" : "Run instances";
```